### PR TITLE
Update docs for a single app server

### PR DESCRIPTION
**Pull request by ipetr0v**
This change updates documentation to represent concepts of a single application server.

Done for #431

Could be built successfully after #585 is merged

### Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by ipetr0v** [comment]
@tiziano88 do you know who created images for docs, and what tools did they use (so the new pictures will have the same style).

**Comment by tiziano88** [comment]
@daviddrysdale did that, though I don't think I know how to re-generate those images from the source files. Perhaps some scripts should be added?

**Comment by daviddrysdale** [comment]
There are [source links](https://github.com/project-oak/oak/blame/master/docs/concepts.md#L115-L116) in the .md files that include the pictures &ndash; they're Google Docs drawings, pulled down with "Download", "PNG Image".

**Comment by ipetr0v** [comment]
WDYT about this new diagram?
BasicFlow seq

**Comment by tiziano88** [comment]
> WDYT about this new diagram?

It's hard to comment about the picture, how about you update the sequence diagram text file and we can comment there?

**Comment by ipetr0v** [comment]
New system diagram may look like this:
SystemDiagram

File: docs/concepts.md [hunk 1]
```diff
@@ -4,10 +4,9 @@
 - [Oak Node](#oak-node)
   - [WebAssembly](#webassembly)
 - [Channels](#channels)
-  - [Pre-Defined Channels and Port Names](#pre-defined-channels-and-port-names)
 - [Pseudo-Nodes](#pseudo-nodes)
 - [Oak Application](#oak-application)
-- [Oak Manager](#oak-manager)
+- [Oak Runner](#oak-runner)
```
[the remark this answers is not in the document]
Comment (daviddrysdale) [reply docs/concepts.md]: (The collection of names on the Oak Module <-> Wasm engine line is also out-of-date, but that's separate from this PR)

File: docs/programming-oak.md [hunk 2]
```diff
@@ -175,118 +176,92 @@ as a gRPC server:
...
+grpc_port: 8080
+initial_node_config_name: "app"
+initial_entrypoint_name: "oak_main"
 ```
 <!-- prettier-ignore-end -->
 
-Alternatively, the manager client can also be wrapped up in a
-[helper library](../oak/client/manager_client.h), which will make application
-configuration (described in the next section) easier:
+The `module_bytes: "<bytes>"` means that this value will be filled with
```
Comment (daviddrysdale) [inline docs/programming-oak.md]: "will be filled" &ndash; by whom?  Is that something that the `app_config_serializer` does?

File: docs/programming-oak.md [hunk 3]
```diff
@@ -175,118 +176,92 @@ as a gRPC server:
...
-### Starting the Oak Application
-
-To use the `CreateApplication` method, the client needs to specify the
-_application configuration_; this describes the Nodes that make up the
-application, and the initial Node to start. However, for the simple case of a
-one-Node application the `oak::ManagerClient` helper takes care of this; the
-client just need to provide the code that is going to be run in the Node, as a
-Wasm module:
+All these steps are implemented as a part of the
+`./scripts/build_example hello_world`.
```
Comment (daviddrysdale) [inline docs/programming-oak.md]: Add "script" or "invocation" before full stop?

File: docs/programming-oak.md [hunk 4]
```diff
@@ -175,118 +176,92 @@ as a gRPC server:
...
 <!-- prettier-ignore-end -->
 
-The Oak Manager will launch an [Oak Runtime](concepts.md#oak-vm) inside the
+The Oak Runner will launch an [Oak Runtime](concepts.md#oak-vm) inside the
 enclave, and this Runtime will check the provided Wasm module(s) and application
 configuration. Assuming everything is correct (e.g. the Nodes all have a main
 entrypoint and only expect to find the Oak
 [host functions](abi.md#host-functions)), the Oak Runtime opens up a port of its
-own and the `CreateApplication` returns this to the client.
+own and returns this to the caller in an `ApplicationCreationStatus`.
```
Comment (daviddrysdale) [inline docs/programming-oak.md]: What is an `ApplicationCreationStatus`, and who's the caller here?

Presumably this contains the host:port that the ISV published to make the Oak App available to the outside world &ndash; be nice to say so.

File: docs/development.md [hunk 5]
```diff
@@ -47,7 +55,7 @@ code as it can help with enabling a faster iteration.
 
 The following command builds and runs an Oak Development Server:
 
-`./scripts/run_server_dev`
+`./scripts/run_server_dev --config="$PWD/bazel-client-bin/examples/hello_world/config/config.bin"`
```
Comment (tiziano88) [inline docs/development.md]: It looks like this `--config` flag may be confused with the `--config=tsan` flag below, but also in general it seems quite generic. Should we consider renaming to `--application` or similar?

File: docs/programming-oak.md [hunk 6]
```diff
@@ -175,118 +176,92 @@ as a gRPC server:
...
   trait.
 
-## Using an Oak Application from a client
+## Starting an Oak Application
 
-A client that is outside of the Oak ecosystem can use an Oak Application by
-interacting with it as a gRPC service. However, this does involve a few
-additional complications beyond the normal gRPC boilerplate, which are described
-in this section.
+In order run an Oak infrastructure ISV should compile an Oak application to a
```
Comment (tiziano88) [inline docs/programming-oak.md]:
```suggestion
In order run an Oak application, the ISV compiles a set of WebAssembly modules, serializes them...
```

File: docs/programming-oak.md [hunk 7]
```diff
@@ -175,118 +176,92 @@ as a gRPC server:
...
+Oak Application.
 
-### Connecting to the Oak Manager
+### Creating a Configuration File
 
-Oak client code (in C++) first needs to connect to the
-[Oak Manager](concepts.md#oak-manager), as a normal gRPC client:
+In order to load an Oak Application into the Oak Server its configuration should
+be serialized into a binary file.
+ISV first need to specify a configuration file:
```
Comment (tiziano88) [inline docs/programming-oak.md]:
```suggestion
The ISV first needs to specify a configuration file:
```

File: docs/programming-oak.md [hunk 8]
```diff
@@ -175,118 +176,92 @@ as a gRPC server:
...
+configuration file and pass it to the Oak Server.
+ISV should also publish a gRPC service endpoint (host:port) for a newly loaded
+Oak Application.
 
-### Connecting to the Oak Manager
+### Creating a Configuration File
 
-Oak client code (in C++) first needs to connect to the
-[Oak Manager](concepts.md#oak-manager), as a normal gRPC client:
+In order to load an Oak Application into the Oak Server its configuration should
```
Comment (tiziano88) [inline docs/programming-oak.md]:
```suggestion
In order to load an Oak Application into the Oak Server its configuration must
```

File: docs/programming-oak.md (hunk 4 above)
Comment (ipetr0v) [reply docs/programming-oak.md]: Done

File: docs/development.md (hunk 5 above)
Comment (ipetr0v) [reply docs/development.md]: Done

File: docs/programming-oak.md (hunk 6 above)
Comment (ipetr0v) [reply docs/programming-oak.md]: Done

File: docs/programming-oak.md (hunk 8 above)
Comment (ipetr0v) [reply docs/programming-oak.md]: Done

File: docs/programming-oak.md (hunk 7 above)
Comment (ipetr0v) [reply docs/programming-oak.md]: Done

File: docs/concepts.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (daviddrysdale) [reply docs/concepts.md]: Perfect, thanks.

File: docs/concepts.md [hunk 9]
```diff
@@ -159,24 +158,21 @@ following system diagram.
 
 ### Workflow
 
-In response to an application creation request, the Oak Manager sends back to
-the caller details about the gRPC endpoint of the newly created Oak Application,
-initialized with the application configuration specified in the request.
+ISV runs an Oak Application using the Oak Runner and publishes details about the
```
Comment (tiziano88) [inline docs/concepts.md]: Question for @daviddrysdale (as native english speaker): should this be "the ISV" or just "ISV" throughout?

Comment (daviddrysdale) [reply docs/concepts.md]: Probably "The ISV"

Comment (ipetr0v) [reply docs/concepts.md]: Done
Also changed all insurances of a "Developer" to the "ISV"

**Pull request merged**